routes/api/habits.js

```diff
@@ -58,14 +58,19 @@ router.post('/',
     (req, res) => {
         Habit.findById(req.params.id).then((habit) => {
             if (habit) {
-                if (req.body.title) {
-                    habit.title = req.body.title;
-                }
+                if (habit.user != req.user.id){
+                    res.status(404).json({ wronguser: 'can only be updated by owner' });
```

Comment:
```suggestion
                    res.status(404).json({ wronguser: 'Habits can only be updated by owner' });
```

[401 unauthorized](https://tools.ietf.org/html/rfc7235#section-3.1) fits better here than [404](https://tools.ietf.org/html/rfc7231#section-6.5.4)